test/dns-server.test.ts

```diff
@@ -0,0 +1,112 @@
+import {
+  createHostedZone,
+  resetHostedZone,
+  createRecord,
+  updateRecord,
+  deleteRecord,
+  listRecords,
+  getChangeStatus,
+  RecordType,
+} from "../app/lib/dns-server";
+
+describe("DNS server lib function test", () => {
+  let hostedZoneId: string;
+
+  beforeAll(async () => {
+    hostedZoneId = await createHostedZone("starchart.com");
+  });
+
+  afterAll(() => resetHostedZone(hostedZoneId));
+
+  test("Hosted zone is created and hosted zone is returned", () => {
+    expect(hostedZoneId.length).toBeGreaterThan(0);
+    expect(/^[A-Z0-9]+/.test(hostedZoneId)).toBe(true);
+  });
+
+  test("Create a record in existing hosted zone but throw an error in non existing hosted zone", async () => {
+    const changeId = await createRecord(
+      RecordType.A,
+      "sub.starchart.com",
+      "192.168.0.1",
+      hostedZoneId
+    );
+
+    expect(changeId).toMatch(/\/change\/[A-Z0-9]+/);
+    await expect(
+      createRecord(
+        RecordType.A,
+        "sub2.starchart.com",
+        "192.168.0.2",
+        "WRONGHOSTEDZONEID"
+      )
+    ).rejects.toThrow();
+  });
+
+  test("Update a record in hosted zone but throw an error in non existing hosted zone", async () => {
```

Comment:
Same comment throughout this file: one thing per test.  Prefer more and smaller test cases.